exporter/sentryexporter/sentry_client.go

```diff
@@ -0,0 +1,314 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sentryexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sentryexporter"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/getsentry/sentry-go"
+)
+
+// sentryAPIClient defines the interface for interacting with Sentry API.
+type sentryAPIClient interface {
+	GetAllProjects(ctx context.Context, orgSlug string) ([]projectInfo, error)
+	GetProjectKeys(ctx context.Context, orgSlug, projectSlug string) ([]projectKey, error)
+	GetOrgProjectKeys(ctx context.Context, orgSlug string) ([]projectKey, error)
+	GetOTLPEndpoints(ctx context.Context, orgSlug, projectSlug string) (*otlpEndpoints, error)
+	CreateProject(ctx context.Context, orgSlug, teamSlug, projectSlug, projectName, platform string) (*projectInfo, error)
+}
+
+// sentryClient handles communication with the Sentry API.
+type sentryClient struct {
+	baseURL   string
+	authToken string
+	client    *http.Client
+}
+
+// newSentryClient is used to override the sentry client factory. While running tests we need to mock the http transport, so that
+// we don't open real sockets.
+var newSentryClient = func(baseURL, authToken string, httpClient *http.Client) sentryAPIClient {
+	return newSentryClientImpl(baseURL, authToken, httpClient)
+}
+
+// projectKey represents a Sentry project key.
+type projectKey struct {
+	ID        string   `json:"id"`
+	Name      string   `json:"name"`
+	Public    string   `json:"public"`
+	Secret    string   `json:"secret"`
+	ProjectID int      `json:"projectId"`
+	IsActive  bool     `json:"isActive"`
+	DSN       dsnField `json:"dsn"`
+}
+
+// dsnField represents the DSN field from API response.
+type dsnField struct {
+	Public string `json:"public"`
+}
+
+// ParsePublicDSN parses the public DSN string into a sentry.Dsn.
+func (d *dsnField) ParsePublicDSN() (*sentry.Dsn, error) {
+	return sentry.NewDsn(d.Public)
+}
+
+// teamInfo represents a Sentry team.
+type teamInfo struct {
+	ID   string `json:"id"`
+	Slug string `json:"slug"`
+	Name string `json:"name"`
+}
+
+// projectInfo represents a Sentry project.
+type projectInfo struct {
+	ID    string     `json:"id"`
+	Slug  string     `json:"slug"`
+	Name  string     `json:"name"`
+	Team  teamInfo   `json:"team"`
+	Teams []teamInfo `json:"teams"`
+}
+
+// otlpEndpoints contains the OTLP endpoint URLs for a project.
+type otlpEndpoints struct {
+	TracesURL string
+	LogsURL   string
+	PublicKey string
+}
+
+// sentryAPIError represents an error from Sentry Management API.
+type sentryAPIError struct {
```

Comment:
We have both `sentryAPIError` and `sentryHTTPError`. Would be nice to share the struct, or make the naming a little more clear that `sentryHTTPError` is for the OTLP endpoint.